src/main/java/org/snomed/snowstorm/ecl/ECLQueryService.java

```diff
@@ -39,8 +43,15 @@
 	}
 
 	public Page<Long> selectConceptIds(String ecl, BranchCriteria branchCriteria, String path, boolean stated, Collection<Long> conceptIdFilter, PageRequest pageRequest) throws ECLException {
+		return selectRelevantConceptIds(ecl, branchCriteria, path, stated, conceptIdFilter, pageRequest, eclEdgeCaseHandlerService.replaceIncorrectConcreteAttributeValue(ecl, path, pageRequest));
+	}
+
+	public Page<Long> selectRelevantConceptIds(String ecl, BranchCriteria branchCriteria, String path, boolean stated, Collection<Long> conceptIdFilter, PageRequest pageRequest, SExpressionConstraint expressionConstraint) {
```

Comment:
This is really the same as `selectConceptIds` but with different arguments. It's more like an internal call because the ECL has already been parsed. Consider naming this `doSelectConceptIds` which is common naming for a lower level call after parsing / processing of the arguments.
I think this method should require the expressionConstraint parameter, otherwise this method could be used to skip the `replaceIncorrectConcreteAttributeValue` behaviour without meaning to. If that is the intention maybe this intension could be reflected in the naming of the method instead?